packages/admin/cms-admin/src/dam/DataGrid/FolderDataGrid.tsx

```diff
@@ -537,14 +543,48 @@ const FolderDataGrid = ({
         },
     ];
 
+    const { data } = useQuery<GQLDamFolderQuery, GQLDamFolderQueryVariables>(damFolderQuery, {
+        variables: {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            id: currentFolderId!,
+        },
+        skip: currentFolderId === undefined,
+    });
+
+    const uploadFilters = {
+        allowedMimetypes: props.allowedMimetypes,
+    };
+
+    function FolderDataGridToolbar() {
```

Comment:
Please don't generate a component during render. Please the component outside of the parent and pass the needed that via props. The useQuery call can be moved into the toolbar